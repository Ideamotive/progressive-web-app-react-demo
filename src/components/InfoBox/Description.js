import React from 'react';
import styled from 'styled-components';


export default styled.div`
    margin: 0 0;
    font-size: 18px;
    line-height: 1.5em;
    padding: 5px 10px 10px;
    text-align: justify;
    border-top: 0;
    `;
