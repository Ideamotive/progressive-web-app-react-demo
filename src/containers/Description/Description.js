import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// Actions
import {
  setPhotoLabel,
} from '../../actions/description';

import {
  pushView,
} from '../../actions/navigation';

// Components
import FullScreenWrapper from '../../components/FullScreenWrapper/FullScreenWrapper';
import PhotoLabel from '../../components/PhotoLabel/PhotoLabel';
import PhotoPreview from '../../components/PhotoPreview/PhotoPreview';
import ActionButton from '../../components/ActionButton/ActionButton';
import ButtonsBar from '../../components/ButtonsBar/ButtonsBar';

export class Description extends Component {
  constructor(props) {
    super(props);
    this.recognition = new (window.SpeechRecognition || window.webkitSpeechRecognition || window.mozSpeechRecognition || window.msSpeechRecognition)();
    this.recognition.lang = 'pl-PL';
    this.recognition.interimResults = false;
    this.recognition.maxAlternatives = 5;
    this.recognition.onresult = this.onRecognitionResult;
  }

  onRecognitionResult = (e) => {
    this.props.setPhotoLabelMainText(e.results[0][0].transcript);
  };

  componentDidMount() {
    if (!this.props.camera.photo) {
      this.props.pushView('CAMERA');
    }
  }

  renderPhotoPreview = (width, height) => <PhotoPreview
    width={width}
    height={height}
    src={this.props.camera.photo || ''}
    isVisible={this.props.transitionState === 'entered'}/>;

  recognitionStart = () => this.recognition.start();
  recognitionStop = () => this.recognition.stop();

  render() {
    return (
      <FullScreenWrapper render={this.renderPhotoPreview}>
        <PhotoLabel
          mainText={this.props.description.photoLabel.mainText}
        />
        <ButtonsBar>
          <ActionButton
            onActionStart={this.recognitionStart}
            onActionEnd={this.recognitionStop}
            iconClass={'fa fa-microphone'}
            danger
          />
          <ActionButton
            onClick={this.props.addNewPhoto}
            iconClass={'fa fa-plus'}
            success
          />

        </ButtonsBar>
      </FullScreenWrapper>
    );
  }
}

Description.propTypes = {
  pushView: PropTypes.func.isRequired,
  transitionState: PropTypes.string.isRequired,
  setPhotoLabelMainText: PropTypes.func.isRequired,
  addNewPhoto: PropTypes.func.isRequired,
  camera: PropTypes.shape({
    photo: PropTypes.string,
  }).isRequired,
  description: PropTypes.shape({
    photoLabel: PropTypes.shape({
      mainText: PropTypes.string.isRequired,
    }),
  }).isRequired,
};

function mapStateToProps(state = {}) {
  const {
    camera,
    description,
  } = state;
  return {
    camera,
    description,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    setPhotoLabelMainText: (...args) => dispatch(setPhotoLabel.mainText(...args)),
    pushView: (...args) => dispatch(pushView(...args)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Description);
