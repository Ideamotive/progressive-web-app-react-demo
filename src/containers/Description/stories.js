import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import { action } from '@storybook/addon-actions';

import { withThemeDecorator } from '../../components/storybook/withThemeDecorators';

import photo from './testPhoto'
import { Description } from './Description';

withThemeDecorator(storiesOf)('Cont. - Description', module)
  .addDecorator(withKnobs)
  .add('Description', () => {
    return (
      <Description
        pushView={action('pushView')}
        transitionState={'entered'}
        setPhotoLabelMainText={action('setPhotoLabelMainText')}
        camera={{ photo: photo }}
        description={{ photoLabel: { mainText: text('Description', 'This is sample photo description') } }}
        addNewPhoto={action('addPhoto')}
      />
    );
  });