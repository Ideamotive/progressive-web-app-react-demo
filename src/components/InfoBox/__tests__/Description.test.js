import React from 'react';
import * as renderer from 'react-test-renderer';

import Description from '../Description';

describe('<Description/> component', () => {
  test('should match to snapshot', () => {
    const tree = renderer
      .create(
        <Description>
          {'test'}
        </Description>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

