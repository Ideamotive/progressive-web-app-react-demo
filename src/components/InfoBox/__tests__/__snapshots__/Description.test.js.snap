// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description/> component should match to snapshot 1`] = `
<div
  className="sc-bdVaJa fWACNT"
>
  test
</div>
`;
